### PR TITLE
chore: update deps and move vue to devDeps, fix #350

### LINKED ISSUE #350: Vue3 version not working due a possible problem with new dom compiler
Seems like latest published version won't work (at least on Vue 3.2) apparently Vue can't match the canvas to its ref
Into `mounted` hook `this.$refs.signaturePadCanvas` is undefined and breaks the component
I get this warning at runtime, tried multiple solution to fix it (using setup and returning render function, combining setup with options api render, etc) but to no avail

image

I recall there was a breaking change to the compiler between Vue 3.1 and 3.2, but rebuilding locally using Vue 3.2 doesn't seem to fix the problem, so I'm pretty lost. This may not impact this library as you're not using `<template>` tags

This problem apparently doesn't affect tests

### PR DESCRIPTION
**Pull request by IlCallo**
Closes #350
Apparently the problem was due to the package installing `vue` as dependency, thus causing some kind of conflict
Moving it to devDeps and re-building the package with Vue 3.2 solved the problem for me, run tests and it's all green

### CODE REVIEW

**Comment by neighborhood999** [comment]
Hi @IlCallo 

Thank you for raising an issue and fix. Can you try to remove `yarn.lock` and re-install dependencies again to confirm this problem?

**Comment by IlCallo** [comment]
I did it multiple times when testing the problem, to no avail

**Comment by IlCallo** [comment]
@neighborhood999 the problem is still present with latest releases
I rebased my branch on latest changes and added vue as peerDependency too

The problem may also be caused by this
image
image

2 version of Vue compiler are present, one required by Vue dep and the one is direct
Vue3 actually serves it's compiler via `vue/compiler-sfc` since 3.2, but apparently the used rollup plugin search for the direct dependency
Vue libraries compiled with compiler version <3.2 aren't compatible with apps using Vue 3.2, the duplicated compiler version in the yarn.lock may be the cause of my issue

Having vue as direct dependency is against best practices and can cause problems, can you test this and merge it if everything still works for you?
Otherwise I'll just republish it with the fix under my company scope

File: package.json [hunk 1]
```diff
@@ -82,7 +81,7 @@
     "@vue/babel-preset-app": "4.5.15",
     "@vue/cli-plugin-babel": "4.5.15",
     "@vue/cli-service": "4.5.15",
-    "@vue/compiler-sfc": "3.1.5",
+    "@vue/compiler-sfc": "^3.2.31",
```
Comment (neighborhood999) [inline package.json:84]: Can you specify the version directly? Same as the below.

```suggestion
    "@vue/compiler-sfc": "3.2.31",
```

**Comment by neighborhood999** [comment]
> @neighborhood999 the problem is still present with latest releases I rebased my branch on latest changes and added vue as peerDependency too
> 
> The problem may also be caused by this image image
> 
> 2 version of Vue compiler are present, one required by Vue dep and the one is direct Vue3 actually serves it's compiler via `vue/compiler-sfc` since 3.2, but apparently the used rollup plugin search for the direct dependency Vue libraries compiled with compiler version <3.2 aren't compatible with apps using Vue 3.2, the duplicated compiler version in the yarn.lock may be the cause of my issue
> 
> Having vue as direct dependency is against best practices and can cause problems, can you test this and merge it if everything still works for you? Otherwise I'll just republish it with the fix under my company scope

Sorry for the reply lately. I thought moving the Vue as the `peerDependencies` is make sense. I left a review, please help change and I will publish the beta version for test.

**Comment by IlCallo** [comment]
Done

**Review by neighborhood999 — APPROVED** [review]
Thanks, will publish the beta version for v3

**Comment by neighborhood999** [comment]
Published `vue-signature-pad@3.0.2-beta.3`, please give it try.

**Comment by IlCallo** [comment]
Yep, it's working
Pay attention, as you're releasing beta packages under the "latest" tag, instead of "beta" one

**Comment by neighborhood999** [comment]
Thank you for testing. You're right, it's my fault, I forgot to add the `beta` tag when I publish.

**Pull request merged**